url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/treemaker/XTreeMaker.java

```diff
@@ -0,0 +1,1486 @@
+				}else if(tokenOk(XTokenKind.KEYWORD, XKeyword.FUNC)){
+					return makeFuncPosStarted(true, XVarAccess.GLOBAL);
+				}
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				expect(XKeyword.SEMICOLON);
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.GLOBAL);
+			}case IF:
+				tokenizer.next();
+				expect(XKeyword.LBRAKET);
+				t = makeExpr();
+				expect(XKeyword.RBRAKET);
+				t2 = makeStatement();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.ELSE)){
+					tokenizer.next();
+					t3 = makeStatement();
+				}
+				return new XTreeIf(endPosition(), (XTreeExpr)t, (XTreeStatement)t2, (XTreeStatement)t3);
+			case IMPORT:
+			{
+				tokenizer.next();
+				List<XTreeImportEntry> imports = makeImports(false);
+				expect(XKeyword.SEMICOLON);
+				return new XTreeImport(endPosition(), imports);
+			}
+			case LOCAL:{
+				tokenizer.next();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.CLASS)){
+					return makeClassPosStarted(true, XVarAccess.LOCAL);
+				}else if(tokenOk(XTokenKind.KEYWORD, XKeyword.FUNC)){
+					return makeFuncPosStarted(true, XVarAccess.LOCAL);
+				}
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				expect(XKeyword.SEMICOLON);
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.LOCAL);
+			}case NONLOCAL:{
+				tokenizer.next();
+				List<XTreeVarDeclEntry> entries = makeDeclEntries();
+				expect(XKeyword.SEMICOLON);
+				return new XTreeVarDecl(endPosition(), entries, XVarAccess.NONLOCAL);
+			}case RETURN:
+				tokenizer.next();
+				if(!tokenOk(XTokenKind.KEYWORD, XKeyword.SEMICOLON))
+					t = makeExpr();
+				expect(XKeyword.SEMICOLON);
+				return new XTreeReturn(endPosition(), (XTreeExpr)t);
+			case SEMICOLON:
+				tokenizer.next();
+				return new XTreeEmpty(endPosition());
+			case SYNCHRONIZED:
+				tokenizer.next();
+				expect(XKeyword.LBRAKET);
+				t = makeExpr();
+				expect(XKeyword.RBRAKET);
+				t2 = makeScope();
+				return new XTreeSynchronized(endPosition(), (XTreeExpr)t, (XTreeScope)t2);
+			case SWITCH:
+				return makeSwitchPosStarted();
+			case THROW:
+				tokenizer.next();
+				t = makeExpr();
+				expect(XKeyword.SEMICOLON);
+				return new XTreeThrow(endPosition(), (XTreeExpr)t);
+			case TRY:
+				return makeTryPosStarted();
+			case WHILE:
+				tokenizer.next();
+				expect(XKeyword.LBRAKET);
+				t = makeExpr();
+				expect(XKeyword.RBRAKET);
+				t2 = makeStatement();
+				return new XTreeWhile(endPosition(), (XTreeExpr)t, (XTreeStatement)t2);
+			case LSCOPE:
+				return makeScope();
+			default:
+				break;
+			}
+			endPosition();
+		}
+		XTreeStatement statement = makeExpr();
+		if(statement instanceof XTreeIdent && tokenOk(XTokenKind.KEYWORD, XKeyword.COLON)){
+			statement = new XTreeLabel(statement.position.clone(), (XTreeIdent)statement, makeStatement());
+		}else{
+			skip = !expect(XKeyword.SEMICOLON);
+		}
+		return statement;
+	}
+	
+	public XTreeStatement makeStatementInteractive() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 155 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 525-533 into a separate method.